cmp/grid/GridModel.js

```diff
@@ -241,7 +248,6 @@ export class GridModel {
         compact,
 
         enableColumnPinning = true,
-        enableColChooser = false,
```

Comment:
I'm on the fence re. not transforming this for developers to make for a smoother transition, but after some thought I think it's the right choice.

However this is a great use for our `apiDeprecated` util to warn if developers are still specifying `enableColChooser`. Otherwise it would be very easy for a dev who doesn't catch the release note to disable column customization in all grids and not necessarily notice it.

I wonder if it would be a better to enable it by default as a core/normal part of Grid. My sense is that we enable in the large majority of grids, and disable only when we know all columns are shown or know we are programatically managing column selection in some other special way.

If we did that we could note it as an additional change and then devs could simply remove the `enableColChooser: true` configs that are likely sprinkled throughout their code.  

Thoughts? Does it seem risky in any way to allow column selection by default? (Especially considering that we *don't* wire up any default persistence).

Reply:
I'm not sure about other apps and their use of col selection, but I can think of a couple places where it would be a bad breaking change - also harder to search the code for the config when it wasn't there to begin with to add `colChooserModel: false`. I like the idea of using apiDeprecated for now to make the transition easier and show exactly where in the code the config needs to be changed.

Reply:
yes, but we mean 'apiRemoved', I think.